dev/tools/shared/node.py

```
@@ -0,0 +1,191 @@
+    else:
+        print(f"Warning: Unsupported architecture for auto-install: {machine}")
+        return None
+
+    version = "v22.22.0"
+    filename = f"node-{version}-{os_name}-{arch}.tar.gz"
+    url = f"https://nodejs.org/dist/{version}/{filename}"
+
+    print(f"Downloading {url}...")
+    tar_path = os.path.join(install_dir, filename)
+    try:
+        urllib.request.urlretrieve(url, tar_path)
```

Ruff (0.15.18)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)